v2/datastream-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/DataStreamToSpanner.java

```diff
@@ -756,4 +810,30 @@ private static DeadLetterQueueManager buildDlqManager(Options options) {
       return DeadLetterQueueManager.create(dlqDirectory, retryDlqUri, 0);
     }
   }
+
+  private static ISchemaOverridesParser configureSchemaOverrides(Options options) {
+    // incorrect configuration
+    if (!options.getSchemaOverridesFilePath().isEmpty()
+        && (!options.getTableOverrides().isEmpty() || !options.getColumnOverrides().isEmpty())) {
+      throw new IllegalArgumentException(
+          "Only one of file based or string based overrides must be configured! Please correct the configuration and re-run the job");
+    }
+    // string based overrides
+    if (!options.getTableOverrides().isEmpty() || !options.getColumnOverrides().isEmpty()) {
```

Comment:
how does this handle if both table and column overrides are non-empty ?

Reply:
That is covered in the parser. If they conflict and the parsing fails, an exception is thrown. Otherwise, both are honored.